test/NCI.OCPL.Api.CTSListingPages.Tests/Tests/Services/ESListingInfoQueryServiceTest.GetByIds.cs

```diff
@@ -0,0 +1,137 @@
+using NCI.OCPL.Api.CTSListingPages.Services;
+
+namespace NCI.OCPL.Api.CTSListingPages.Tests
+{
+    public partial class ESListingInfoQueryServiceTest
+    {
+        /// <summary>
+        /// Test to verify that Elasticsearch requests are being assembled correctly.
+        /// </summary>
+        [Fact]
+        public async void GetByIds_TestRequestSetup()
+        {
+            string[] codes = new string[] { "C115270", "C8578", "C9092", "C3017" };
+            JObject expectedRequest = JObject.Parse(
+@"{
+    ""query"": {
+        ""terms"": { ""concept_id"": [ ""C115270"", ""C8578"", ""C9092"", ""C3017"" ] }
+    }
+}
+");
+
+            Uri esURI = null;
+            string esContentType = String.Empty;
+            HttpMethod esMethod = HttpMethod.DELETE; // Basically, something other than the expected value.
+
+            JObject requestBody = null;
+
+            ElasticsearchInterceptingConnection conn = new ElasticsearchInterceptingConnection();
+            conn.RegisterRequestHandlerForType<Nest.SearchResponse<ListingInfo>>((req, res) =>
+            {
+                // We don't really care about the response for this test.
+                res.Stream = MockEmptyResponse;
+                res.StatusCode = 200;
+
+                esURI = req.Uri;
+                esContentType = req.ContentType;
+                esMethod = req.Method;
+                requestBody = conn.GetRequestPost(req);
+            });
+
+            // The URI does not matter, an InMemoryConnection never requests from the server.
+            var pool = new SingleNodeConnectionPool(new Uri("http://localhost:9200"));
+
+            var connectionSettings = new ConnectionSettings(pool, conn);
+            IElasticClient client = new ElasticClient(connectionSettings);
+
+            // Setup the mocked Options
+            IOptions<ListingPageAPIOptions> clientOptions = GetMockOptions();
+
+            ESListingInfoQueryService query = new ESListingInfoQueryService(client, clientOptions, new NullLogger<ESListingInfoQueryService>());
+
+            // For this test, we don't really care that this returns anything, only that the intercepting connection
+            // sets up the request correctly.
+            await query.GetByIds(codes);
+
+            Assert.Equal("/listingpagev1/ListingInfo/_search", esURI.AbsolutePath);
+            Assert.Equal("application/json", esContentType);
+            Assert.Equal(HttpMethod.POST, esMethod);
+            Assert.Equal(expectedRequest, requestBody, new JTokenEqualityComparer());
+        }
+
+        public static IEnumerable<object[]> GetByIds_Scenarios = new[]
+        {
+            //new object[] { new GetByIds_NoResults() },
+            //new object[] { new GetByIds_SingleResult() },
+            new object[] { new GetByIds_MultipleResults() }
+        };
+
+        /// <summary>
+        /// Test to verify handling when no results are returned.
+        /// </summary>
```

Comment:
The comment for this test doesn't match with what the test data claims to be.

Reply:
Updated the comment here, and in GetByPrettyUrlName tests.